### PR TITLE
add erc-7509

### PR DESCRIPTION
**Pull request by HelloRickey**
It is migrated from https://github.com/ethereum/EIPs/pull/7651

### CODE REVIEW

File: ERCS/erc-7509.md [hunk 1]
```diff
@@ -0,0 +1,329 @@
...
+- It can help your application add new features or upgrades, because data and behavior are separated, new features will not affect your old data.
+- It is easy to manage. When your application consists of multiple contracts, it will help you effectively manage the status of each contract.
+- Its components are reusable, and you can share your components with the community to help others improve development efficiency.
+
+## Specification
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "NOT RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in RFC 2119 and RFC 8174.
+
+World contracts are containers for entities, component contracts, and system contracts. Its core principle is to establish the relationship between entities and component contracts, and different entities will attach different components. And use the system contract to dynamically change the data of the entity in the component.
+Usual workflow when building ECS-based programs
```
Comment (SamWilsn) [inline ERCS/erc-7509.md:40]: The wording and grammar are a bit off here. Does this get the correct point across?

```suggestion
World contracts are containers for entities, component contracts, and system contracts. Its core principle is to establish the relationship between entities and component contracts, where different entities will attach different components, and use system contracts to dynamically change the data of the entity in the component.

Usual workflow when building ECS-based programs:
```

File: ERCS/erc-7509.md [hunk 2]
```diff
@@ -0,0 +1,329 @@
...
+    function get(uint256 _entityId, bytes memory _params)
+        external
+        view
+        returns (bytes memory);
+}
+```
+
+### Library
+
+The library [`Types.sol`](../assets/erc-7509/Types.sol) contains an enumeration of Solidity types used in the below interfaces.
```
Comment (SamWilsn) [inline ERCS/erc-7509.md:280]:
```suggestion
The library [`Types.sol`](../assets/erc-7509/Types.sol) contains an enumeration of Solidity types used in the above interfaces.
```

**Review by SamWilsn — DISMISSED** [review]
I think this is good to go for draft. Just make sure to fix the two comments at some point.

**Comment by HelloRickey** [comment]
Hey @SamWilsn , your comments has been fixed and all checks have passed. Please do a final check. Thank you very much.

**Comment by HelloRickey** [comment]
@SamWilsn I have addressed all required feedback and ready for a review.

**Comment by HelloRickey** [comment]
@SamWilsn  The merge was interrupted because of this issue https://github.com/ethereum/ERCs/issues/110

**Pull request merged**